OsmAnd/src/net/osmand/plus/plugins/monitoring/OsmandMonitoringPlugin.java

```diff
@@ -259,6 +183,38 @@ public void mapActivityPause(MapActivity activity) {
 		this.mapActivity = null;
 	}
 
+	@Override
+	public void createWidgets(@NonNull MapActivity mapActivity, @NonNull List<MapWidgetInfo> widgetsInfos, @NonNull ApplicationMode appMode) {
+		MapWidgetRegistry widgetRegistry = app.getOsmandMap().getMapLayers().getMapWidgetRegistry();
+
+		MapWidget distanceWidget = createMapWidgetForParams(mapActivity, TRIP_RECORDING_DISTANCE);
```

Comment:
Looks like MapWidget should contain WidgetParams.
WidgetParams should be renamed to WidgetType